uitools/widgets/Esri/ArcGISRuntime/Toolkit/BasemapGallery.cpp

```diff
@@ -0,0 +1,54 @@
+#include "BasemapGallery.h"
+#include "ui_BasemapGallery.h"
+
+
+//Toolkit controllers
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+BasemapGallery::BasemapGallery(GeoModel* geomodel, QWidget *parent) :
+  QFrame(parent),
+  m_ui(new Ui::BasemapGallery), m_controller(new BasemapGalleryController(this))
+{
+  m_ui->setupUi(this);
+  //setting the controller basemap and geoview
+  m_controller->setCurrentBasemap(geomodel->basemap());
+  m_controller->setGeoModel(geomodel);
+
+  GenericListModel* model = m_controller->gallery();
+  m_ui->listView->setModel(model);
+  /*have to set the property names, so the controller will know how to match the properties from
+  basemapgalleryitem with the specific Qt::<namespace> invoked in the .data() from the View (ListView) obj*/
+  model->setDisplayPropertyName("name");
+  model->setDecorationPropertyName("thumbnail");
+  model->setTooltipPropertyName("tooltip");
```

Comment:
I would move these to the `BasemapGalleryController`. They make sense in that context I think.